torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -286,11 +320,38 @@ class AttributePropagator {
               paramConst->debugName());
           n->outputs().at(0)->replaceAllUsesWith(paramConst);
           n->removeAllInputs();
+        } else if (n->kind() == prim::fork) {
+          auto func = [this](std::shared_ptr<Graph>& subgraph) {
+            propagateAttributes(subgraph);
+          };
+          applyToSubGraph(n, graph, func);
         }
       }
     }
   }
 
+  void applyToSubGraph(
+      Node* n,
+      std::shared_ptr<Graph>& graph,
+      const std::function<void(std::shared_ptr<Graph>&)>& func) {
```

Comment:
It might be easier to follow if this took in an enum of `recordMutableAttrs, recordReferencedAttrs, propagateAttributes` instead of a lambda